lib/msf/core/exploit/remote/http/manage_engine_adaudit_plus/target_info.rb

```diff
@@ -0,0 +1,259 @@
+    configured_domains = []
+    begin
+      domain_info = JSON.parse(res.body)
+      if domain_info && domain_info.include?('domainFullList') && !domain_info['domainFullList'].empty?
+        domain_full_list = domain_info['domainFullList']
+        domain_full_list.each do |domain|
+          next unless domain.is_a?(Hash) && domain.key?('name')
+
+          domain_name = domain['name']
+          next if domain_name.empty?
+
+          configured_domains << domain_name
+        end
+      else
+        print_error('Failed to identify any configured domains.')
+      end
+    rescue JSON::ParserError => e
+      print_error('Failed to identify any configured domains - The server response did not contain valid JSON.')
+      print_error("Error was: #{e.message}")
+    end
+
+    if configured_domains.empty?
+      return {
+        'status' => adaudit_plus_status::NO_MATCH,
+        'message' => 'Failed to obtain the list of configured domains.',
+        'adapcsrf_cookie' => adapcsrf_cookie.value
+      }
+    end
+
+    print_status("Found #{configured_domains.length} configured domain(s): #{configured_domains.join(', ')}")
+    # we don't need to return a message here
+    {
+      'status' => adaudit_plus_status::SUCCESS,
+      'adapcsrf_cookie' => adapcsrf_cookie.value,
+      'configured_domains' => configured_domains
+    }
+  end
+
+  # Check the build number for the ADAudit Plus installation
+  #
+  # @param adapcsrf_cookie [String] A valid ADAP CSRF cookie for API calls.
+  # @see adaudit_plus_login The function which can be called to obtain a
+  #   valid CSRF cookie that can be used by this code.
+  # @return [Hash] Hash containing a `status` key, which is used to hold a
+  #   status value as an Integer value, a `message` key, which is used
+  #   to hold a message associated with the status value as a String,
+  #   and an optional 'build_version' key, which is used to hold an object
+  #   of type Reg::Version if the build number was successfully obtained.
+  def adaudit_plus_grab_build(adapcsrf_cookie)
+    vprint_status('Attempting to obtain the ADAudit Plus build number')
+
+    res = send_request_cgi({
+      'uri' => adaudit_plus_license_details_uri,
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'vars_post' => { 'adapcsrf' => adapcsrf_cookie.to_s }
+    })
+
+    unless res
+      return {
+        'status' => adaudit_plus_status::CONNECTION_FAILED,
+        'message' => 'Connection failed while attempting to obtain the build number.'
+      }
+    end
+
+    unless res.code == 200
+      return {
+        'status' => adaudit_plus_status::UNEXPECTED_REPLY,
+        'message' => "Received unexpected HTTP response #{res.code} when attempting to obtain the build number."
+      }
+    end
+
+    build = res.body.scan(/"buildNumber":"(\s{0,}\d{4}\s{0,})",/)&.flatten&.first
+    if build.blank?
+      return {
+        'status' => adaudit_plus_status::NO_MATCH,
```

Comment:
This should really be its own error code such as `adaudit_plus_status::NO_BUILD_NUM`

Reply:
agreed, fixed